inputgraph/inputgraph_test.go

```diff
@@ -0,0 +1,51 @@
+package inputgraph
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/earthly/earthly/conslogging"
+	"github.com/earthly/earthly/domain"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHashTargetWithDocker(t *testing.T) {
+	r := require.New(t)
+	target := domain.Target{
+		LocalPath: "./testdata/with-docker",
+		Target:    "with-docker-load",
+	}
+
+	ctx := context.Background()
+	cons := conslogging.New(os.Stderr, &sync.Mutex{}, conslogging.NoColor, 0, conslogging.Info)
+
+	org, project, hash, err := HashTarget(ctx, target, cons)
+	r.NoError(err)
+	r.Equal("earthly-technologies", org)
+	r.Equal("core", project)
+
+	hex := fmt.Sprintf("%x", hash)
+	r.Equal("b644ef18314c4d28c98a26ff21c593ac25a21c86", hex)
```

Comment:
can you expand the test to validate that making a change to the Earthfile produces a different hash?

e.g. if a user changed `SAVE IMAGE saved:latest` to `SAVE IMAGE saved:newer` it would produce a different hash value.